pkgs/development/arduino/platformio/core.nix

```diff
@@ -47,25 +49,25 @@ let
 
 in buildPythonApplication rec {
   pname = "platformio";
-  version = "4.0.3";
+  version = "4.1.0";
 
   # pypi tarballs don't contain tests - https://github.com/platformio/platformio-core/issues/1964
   src = fetchFromGitHub {
     owner = "platformio";
     repo = "platformio-core";
     rev = "v${version}";
-    sha256 = "1naaa53cc7n7zyqggqjvvgkcq8cyzngdf904y9ag0x1vvb70f8j9";
+    sha256 = "10v9jw1zjfqr3wl6kills3cfp0ky7xbm1gc3z0n57wrqbc6cmz95";
   };
 
   propagatedBuildInputs =  [
     bottle click colorama git lockfile
     pyserial requests semantic-version
-    tabulate
+    tabulate pyelftools marshmallow2
```

Comment:
You can use this patch to use marshmallow3: https://github.com/NixOS/nixpkgs/pull/76742/files#diff-49c02924ff146dbf06ebbdb056ae1759R82

Reply:
This is much preferred to having an incoherent python package set.

Reply:
Sorry, didn't notice your PR, @Mic29. Thanks! I'll incorporate your changes.